### ISSUE TITLE
[Google Maps API] Inconsistent results searching with phone number

### ISSUE THREAD

**Issue by alexbarron** [serpapi/public-roadmap#1842]
A customer reported a difference in how our Google Maps API handles phone number searches compared to actual Google Maps.

You can search Google Maps with a phone number and you can get the business associated with that number.

For example searching for `+44 1223 870791`, finds the Royal Oak pub.

https://www.google.com/maps/search/%2B44+1223+870791/@51.8619833,0.1396743,7z?hl=en

![image](https://github.com/user-attachments/assets/9926766b-b2fc-4880-8cf3-1bb3692a73a3)

However, the same search in our Google Maps API returns no results.

[Playground](https://serpapi.com/playground?engine=google_maps&q=%2B44+1223+870791&ll=%4051.8619833%2C0.1396743%2C7z&hl=en&type=search)

![image](https://github.com/user-attachments/assets/e0f242c6-6222-45e0-b057-f36c025d8a83)

I confirmed the same behavior with other numbers in other countries such as `+39 06 581 2260`.

[Playground](https://serpapi.com/playground?engine=google_maps&q=%2B39+06+581+2260&ll=%4041.8880887%2C12.4752751%2C7z&hl=en&type=search)

Interestingly, this number `+44 1279 836303` in our Google Maps API returned many results of businesses with phone numbers beginning with `+44 1279`, but the actual business with this number was in position 13. See Harvest Moon on the screenshot below.

[Playground](https://serpapi.com/playground?engine=google_maps&q=%2B39+06+581+2260&ll=%4041.8880887%2C12.4752751%2C7z&hl=en&type=search)

![image](https://github.com/user-attachments/assets/5bbfd80e-a0d4-41a7-bdb3-cefb8aceee39)

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/conversation/102910000069498?view=List)

**Comment by aciddjus** [comment]
> Interestingly, this number `+44 1279 836303` in our Google Maps API returned many results of businesses with phone numbers beginning with `+44 1279`, but the actual business with this number was in position 13. See Harvest Moon on the screenshot below.

Thank you, @alexbarron. Can you please post the Inspect link for this searach?

**Comment by alexbarron** [comment]
Of course, here's the [Inspect link](https://serpapi.com/searches/66bdc3236f9f6349c8b240c2/inspect). Thank you for looking into this.

**Comment by alexbarron** [comment]
> I was able to get results locally with only one of our proxy providers, with a success rate of about 10%.
> <img alt="image" width="1897" src="https://private-user-images.githubusercontent.com/6929257/358226270-c7b5d0ed-d7ed-4a5e-8cd5-58868e745e6c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3Mjk0OTIsIm5iZiI6MTcyMzcyOTE5MiwicGF0aCI6Ii82OTI5MjU3LzM1ODIyNjI3MC1jN2I1ZDBlZC1kN2VkLTRhNWUtOGNkNS01ODg2OGU3NDVlNmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTVUMTMzOTUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjQyYjFmYTBlYzY5YjU0NTgyMzIxMDVmOGJmNDNiMWU3MDAwYjdhN2UxYjUwYWUxZmY2NGNhZmQ5ZjdkMDBlNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.GhYCXCCTlO-OGp_KDOT17X0NXBV15Vnlz2AAp_ycZ5s">

That makes sense actually. The customer who reported this got lucky with a phone number and managed to get the right result once through our API. I couldn't reproduce it on my end with the same exact params.

**Comment by ocvit** [comment]
The issue is not reproducible anymore. Both examples ([1](https://serpapi.com/playground?engine=google_maps&q=%2B44+1223+870791&ll=%4051.8619833%2C0.1396743%2C7z&hl=en&type=search&no_cache=true), [2](https://serpapi.com/playground?engine=google_maps&q=%2B39+06+581+2260&ll=%4041.8880887%2C12.4752751%2C7z&hl=en&type=search&no_cache=true)) reliably return correct place results.

<img width="1490" height="1217" alt="Image" src="https://github.com/user-attachments/assets/84cb25c2-07ed-4711-a064-3cf1a9ea1bea" />
